[fr] Move new observability primer page

iguitton opened the pull request:
- Fixes #5913
- Moves observability-primer.md under fr/docs/concepts

dmathieu commented:
It looks like there are now hash issues in links (see the failing GH Action).

chalin commented on content/fr/_index.md:
```diff
@@ -57,7 +57,7 @@ performance et le comportement de vos logiciels.
 {{% blocks/section color="dark" type="row" %}}
 
-{{% blocks/feature icon="fas fa-chart-line" title="Traces, Métriques, Logs" url="docs/concepts/observability-primer/" %}}
+{{% blocks/feature icon="fas fa-chart-line" title="Traces, Métriques, Logs" %}}
```
Why this change? It is forcing a difference between the `fr` and `en` pages (which shouldn't be necessary) and invalidating the `default_lang_commit` value.

iguitton replied:
@chalin : I was struggling to solve the link issue in the _index.md. I checked how other languages handled this and noticed that both the Spanish and Portuguese versions removed this URL.

chalin commented on content/fr/docs/what-is-opentelemetry.md:
Similar to my other comment, the changes made here need to be undone.

chalin commented:
The "hash does not exist" link-check failures of this PR are because `content/fr/docs/what-is-opentelemetry.md` doesn't have heading IDs, as recommended in [Localization > Heading anchors](https://opentelemetry.io/docs/contributing/localization/#heading-anchors). The proper fix is here is to add the heading IDs in `content/fr/docs/what-is-opentelemetry.md`.

chalin commented:
> * Fixes [[fr] Move new observability-primer page under `docs` #5913](https://github.com/open-telemetry/opentelemetry.io/issues/5913)
> * Moves observability-primer.md under fr/docs/concepts

In passing, I'd like to suggest that you add "Fixes #XXXX" to the opening comment of a PR rather than the title. By doing so you allow GitHub to automatically link the PR to the issue. This make is easier for all to navigate to the issue, and allows GitHub to know that the issue can be closed when the PR is merged.

iguitton commented:
@chalin, I’ve reverted the changes and added heading IDs to what-is-opentelemetry.md, but the issue persists. Could you advise on the next steps? I’m unsure why this is still failing and would appreciate your guidance.

chalin commented:
> @chalin, I’ve reverted the changes and added heading IDs to what-is-opentelemetry.md, but the issue persists. Could you advise on the next steps? I’m unsure why this is still failing and would appreciate your guidance.

Hi @iguitton. There are two issues:

(1) As mentioned in #5913, you need to **move** the new observability-primer page under docs. This PR just creates a new file under docs, but the old one is still there.

(2) As reported in https://github.com/open-telemetry/opentelemetry.io/actions/runs/12825801302/job/35764655095?pr=5927: 

```text
fr/docs/index.html
  hash does not exist --- fr/docs/index.html --> concepts/observability-primer/#what-is-observability
```

Your new page `fr/docs/concepts/observability-primer` still doesn't have the heading ID `what-is-observability`, as you can see here:

> <img width="681" alt="image" src="https://github.com/user-attachments/assets/150710a9-cdc0-4f75-9ec1-a4cb8cf0f406" />

You did add the heading ID to `content/fr/docs/what-is-opentelemetry.md`, but it is missing from `observability-primer.md`.

My suggestion, to avoid such issues in the future, is to add heading IDs for all headings as you translate new pages, rather than wait later for the link checker to report errors.

chalin commented on content/fr/docs/what-is-opentelemetry.md:
```diff
@@ -8,10 +8,9 @@ default_lang_commit: 71833a5f8b84110dadf1e98604b87a900724ac33
-  d'[observabilité](/docs/concepts/observability-primer/#what-is-observability)
-  et des outils conçus pour créer et gérer des données de télémétrie telles que
-  des [traces](/docs/concepts/signals/traces/), des
+- Un framework d'[observabilité](/docs/concepts/observability-primer/) et des
+  outils conçus pour créer et gérer des données de télémétrie telles que des
+  [traces](/docs/concepts/signals/traces/), des
```
This, and similar changes in this page that remove `#what-is-observability` from links, need to be undone.

iguitton commented:
/fix:format

iguitton commented:
/fix:format

iguitton commented:
/fix:all

chalin reviewed (approved):
LGTM 🎉

chalin commented on content/fr/docs/concepts/observability-primer.md:
```diff
@@ -96,7 +96,7 @@ lorsqu'ils sont mis en corrélation avec une trace ou un span.
 référez-vous à la page [Logs](/docs/concepts/signals/logs/).
 
-### Spans
+### Spans {#spans}
```
@iguitton - you don't actually need the ID here since the heading text corresponds to `#spans`, so maybe consider dropping the ID.

chalin replied:
If you decide to remove it, consider removing it for the `### Logs` heading as well.

iguitton commented:
> LGTM 🎉

Thanks for your patience and help, @chalin—it is much appreciated! 😊

chalin commented:
Waiting on an approval from @open-telemetry/docs-fr-approvers
